hermes/bin/src/wasm/context.rs

```diff
@@ -0,0 +1,6 @@
+//! WASM host context implementation.
+
+/// A WASM host context structure, which is intended to be passed to the `wasmtime::Store`
+/// during the WASM state initialization process.
+#[derive(Clone)]
+pub(crate) struct Context;
```

Comment:
You can make a "general" instance.  and make `new` which creates it.
Then you can have a method on it called `use_for(event:string)` which increments the Execution Count in the base instance, then clones the base instance and sets the `Event name` field to the parameter.
This then returns a "usable" context which we would attach to the store.

The reason I want these two extra parameters, is so when we call functions in the runtime, we can check is the function being called by the same "event" within the module, and also, is it the same "execution".  it would allow the extension to control if its state can be shared between executions or is bounded by a particular execution.

The choice and how these fields get used is up to the runtime extension, but with these 4, we have pretty fine granularity over managing the external state accurately.

Reply:
I would use this for the execution counter:
https://doc.rust-lang.org/std/sync/atomic/struct.AtomicU64.html
As it will be thread safe,  and should be high performance.